rustls/tests/api.rs

```diff
@@ -1349,6 +1349,55 @@ fn client_cert_resolve_default() {
     }
 }
 
+#[test]
+fn client_cert_resolve_server_no_hints() {
+    // Test that a server can provide no hints and the client cert resolver gets the expected
+    // arguments.
+    for key_type in ALL_KEY_TYPES.into_iter() {
+        // Build a verifier with no hint subjects.
+        let verifier = WebPkiClientVerifier::builder(get_client_root_store(key_type))
+            .clear_root_hint_subjects();
+        let server_config = make_server_config_with_client_verifier(key_type, verifier);
+        let expected_root_hint_subjects = Vec::default(); // no hints expected.
+        test_client_cert_resolve(key_type, server_config.into(), expected_root_hint_subjects);
+    }
+}
+
+#[test]
+fn client_cert_resolve_server_added_hint() {
+    // Test that a server can add an extra subject above/beyond those found in its trust store
+    // and the client cert resolver gets the expected arguments.
+    let extra_name = b"0\x1a1\x180\x16\x06\x03U\x04\x03\x0c\x0fponyland IDK CA".to_vec();
+    for key_type in ALL_KEY_TYPES.into_iter() {
+        let expected_hint_subjects = match key_type {
+            KeyType::Rsa => {
+                vec![
```

Comment:
I think there's a fair bit of low-hanging deduplication fruit here?

Reply:
Tidied this up.